From f6ac9b7b7c5417588f99ac959716b47bbd7140a5 Mon Sep 17 00:00:00 2001
From: Jookia <contact@jookia.org>
Date: Sun, 3 Dec 2023 00:34:32 +1100
Subject: [PATCH] Fix Makefile for Buildroot building

---
 Makefile | 44 ++++++++++++++++++++++----------------------
 1 file changed, 22 insertions(+), 22 deletions(-)

diff --git a/Makefile b/Makefile
index 7b4a831..012ad83 100644
--- a/Makefile
+++ b/Makefile
@@ -4,9 +4,9 @@ SHELL = /bin/sh
 
 .SUFFIXES: .c .o
 
-prefix ?= ./usr/local
+PREFIX ?= /usr/local
 
-INSTALL = /usr/bin/install -c
+INSTALL = install
 INSTALL_PROGRAM = $(INSTALL)
 INSTALL_DATA = $(INSTALL) -m 644
 RM = rm -f
@@ -33,44 +33,44 @@ LIBS    = $(LIB_BMI088) $(LIB_AKICM)
 # $(warning srcdir=$(srcdir))
 VPATH = $(srcdir)/src:$(srcdir)/bosch-lib
 
-CPPFLAGS   = -I. -I$(srcdir)/src -I$(srcdir)/bosch-lib -DBMI08X_ENABLE_BMI085=0 -DBMI08X_ENABLE_BMI088=1
-CFLAGS     = -g
-ALL_CFLAGS = $(CPPFLAGS) $(CFLAGS)
+EXTRAFLAGS = -I$(srcdir)/bosch-lib -DBMI08X_ENABLE_BMI085=0 -DBMI08X_ENABLE_BMI088=1
+ALL_CFLAGS = $(CFLAGS) $(EXTRAFLAGS)
 
 all: $(TARGETS) $(LIBS)
 
-# option -Wl,--rpath=./ used by bmi088_test under developing environment
+%.o: %.c
+	$(CC) $(ALL_CFLAGS) -c -o $@ $^
+
 $(TST_BMI088): test_bmi088.o $(LIB_BMI088)
-	$(CC)  $(ALL_CFLAGS) -o $@ -L./ -Wl,-\( -lbmi088 -Wl,--rpath=./ $< -Wl,-\)
+	$(CC) -o $@ $^
 
 $(TST_ICM20600): test_icm20600.o $(LIB_AKICM)
-	$(CC)  $(ALL_CFLAGS) -o $@ -L./ -Wl,-\( -lakicm -Wl,--rpath=./ $< -Wl,-\)
+	$(CC) -o $@ $^
 
 $(TST_AK09918): test_ak09918.o $(LIB_AKICM)
-	$(CC)  $(ALL_CFLAGS) -o $@ -L./ -Wl,-\( -lakicm -Wl,--rpath=./ $< -Wl,-\)
+	$(CC) -o $@ $^
 
 $(LIB_BMI088): $(OBJS_BMI088)
-	$(CC)  $(ALL_CFLAGS) --shared -o $@ $^
+	$(CC) --shared -o $@ $^
 
 $(LIB_AKICM): $(OBJS_AKICM)
-	$(CC)  $(ALL_CFLAGS) --shared -o $@ $^
+	$(CC) --shared -o $@ $^
 
 install: all
-	$(INSTALL) -D $(TST_BMI088) $(DESTDIR)$(prefix)/bin/$(TST_BMI088)
-	$(INSTALL) -D $(TST_ICM20600) $(DESTDIR)$(prefix)/bin/$(TST_ICM20600)
-	$(INSTALL) -D $(TST_AK09918) $(DESTDIR)$(prefix)/bin/$(TST_AK09918)
-	$(INSTALL) -D $(LIB_BMI088) $(DESTDIR)$(prefix)/lib/$(LIB_BMI088)
-	$(INSTALL) -D $(LIB_AKICM) $(DESTDIR)$(prefix)/lib/$(LIB_AKICM)
+	$(INSTALL) -D $(TST_BMI088) $(DESTDIR)$(PREFIX)/bin/$(TST_BMI088)
+	$(INSTALL) -D $(TST_ICM20600) $(DESTDIR)$(PREFIX)/bin/$(TST_ICM20600)
+	$(INSTALL) -D $(TST_AK09918) $(DESTDIR)$(PREFIX)/bin/$(TST_AK09918)
+	$(INSTALL) -D $(LIB_BMI088) $(DESTDIR)$(PREFIX)/lib/$(LIB_BMI088)
+	$(INSTALL) -D $(LIB_AKICM) $(DESTDIR)$(PREFIX)/lib/$(LIB_AKICM)
 
 uninstall:
-	-$(RM) $(DESTDIR)$(prefix)/bin/$(TST_BMI088)
-	-$(RM) $(DESTDIR)$(prefix)/bin/$(TST_AK09918)
-	-$(RM) $(DESTDIR)$(prefix)/bin/$(TST_ICM20600)
-	-$(RM) $(DESTDIR)$(prefix)/lib/$(LIB_BMI088)
-	-$(RM) $(DESTDIR)$(prefix)/lib/$(LIB_AKICM)
+	-$(RM) $(DESTDIR)$(PREFIX)/bin/$(TST_BMI088)
+	-$(RM) $(DESTDIR)$(PREFIX)/bin/$(TST_AK09918)
+	-$(RM) $(DESTDIR)$(PREFIX)/bin/$(TST_ICM20600)
+	-$(RM) $(DESTDIR)$(PREFIX)/lib/$(LIB_BMI088)
+	-$(RM) $(DESTDIR)$(PREFIX)/lib/$(LIB_AKICM)
 
 clean:
 	-$(RM) *.o $(TARGETS) $(LIBS)
 
 .PHONY: all clean install uninstall
-
-- 
2.42.1

